homeassistant/components/suez_water/sensor.py

```diff
@@ -24,46 +24,76 @@ async def async_setup_entry(
 ) -> None:
     """Set up Suez Water sensor from a config entry."""
     coordinator = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities([SuezAggregatedSensor(coordinator, entry.data[CONF_COUNTER_ID])])
+    counter_id = entry.data[CONF_COUNTER_ID]
+    async_add_entities(
+        [
+            SuezAggregatedSensor(coordinator, counter_id),
+            SuezPriceSensor(coordinator, counter_id),
+        ]
+    )
 
 
-class SuezAggregatedSensor(CoordinatorEntity[SuezWaterCoordinator], SensorEntity):
-    """Representation of a Sensor."""
-
+class _SuezSensor(CoordinatorEntity[SuezWaterCoordinator], SensorEntity):
```

Comment:
Considered using entity descriptions to allow new entities to be added quicker and with less code?

Reply:
Due to the extra state defined in SuezAggregatedSensor it does not seems to be possible.  
But I might be wrong as I'm not completely sure of how entity descriptions works.

Reply:
A common pattern is that you extend the `SensorEntityDescription` with extra functions like `value_fn` or `attributes_fn`. This way you can extend the behaviour

Reply:
done 😄 

Not much gain in code for now, but I have seen your integration and I understand it.
But this is more readable so it's nice.